scijava/scijava-ops-api/templates/main/java/org/scijava/ops/api/OpBuilder.vm

```diff
@@ -47,13 +47,52 @@ import org.scijava.types.Types;
 
 /**
  * Convenience class for looking up and/or executing ops using a builder
- * pattern.
+ * pattern. The general order of specification:
+ * <ol>
+ *   <li>The op name</li>
+ *   <li>The number of input(s)</li>
```

Comment:
I love the additional level of detail here! But it gives me an idea to make the builder more succinct: the new `arityX` methods given subsequent to `op(opName)` do not take any arguments, right? So what if we got rid of `op(String)` in favor of directly asking for the arity from the `OpEnvironment`? So we'd have something like:

```java
ops.unary("math.sqrt").input(5).outType(Double.class).apply()
ops.binary("math.pow").inputs(6, 3).output(myDoubleTypeResult).compute()
ops.arity3("logic.ternary").inputs(myConditionalValue, myTrueValue, myFalseValue).apply()
```
and so forth.

@hinerm What do you think?

Reply:
Relatedly: this javadoc would be more accessible if put onto the `.op(String)` method of the `OpEnvironment`, don't you think? That's where people will be discovering things via autocomplete. Is it likely they are going to stumble upon the javadoc for the `OpBuilder` class itself, when it is usually not imported and referenced directly?

Reply:
@hinerm Some thoughts in response to your thoughts:

* We could put both `unary`/`binary` and `arity1`/`arity2` so they're easier to find? Or if the redundancy is too yucky, I I guess I lean toward `unary`/`binary`; we already special-case the `Function` and `BiFunction` because they already exist and creating `Functions.Arity1` and `Functions.Arity2` felt redundant, so in that way, `unary`/`binary` is more consistent naming-wise.
* For the `Hints` support... I think wanting to pass a `Hints` object is in the small minority of cases. So what if we *kept* `op(String)` and `op(String, Hints)` signatures, but *also* added `unary(String)` as a default interface method that calls `op(String).arity1()` under the hood, and `binary(String)` calling `op(String).arity2()`, etc.? Then people have the nicest + shortest + most readable way, but also the way to specify hints that is still pretty elegant and nice to read.
* For the auto-generation: my vote would be to just add the 16-17 needed default method one-liners by hand and not worry about autogenerating them. If we ever change the number of built-in arities, we can just update the `OpEnvironment` manually as well.

Reply:
One other way of handling the hints would be to change all the various builder classes to extend a common `OpBuilderStep<T extends OpBuilderStep<T>>` base class, which has a `public T hints(Hints hints)` method that stores the hints into a protected hints variable. So then you could chain-add your `Hints` object at *any* step of the build.

Reply:
I'm liking the use of `unary` and `binary` from the `OpEnvironment`, while keeping it `arity1()` and `arity2()` from the `OpBuilder`, more and more as it settles in my brain. You can even cross-link them in the javadoc, where the javadoc for `arity1()` says something like `@see OpEnvironment#unary for a shorthand way to access this path` and similar for binary.